akka-docs/src/main/paradox/testing-typed.md

```diff
@@ -29,4 +29,133 @@ Maven
+
+For creating a child actor a noop actor is created:
+
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #child }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #child }
+
+All of the tests make use of the `BehaviourTestkit` to avoid the need for a real `ActorContext`. Some of the tests
+make use of the `TestInbox` which allows the creation of an `ActorRef` that can be used for synchronous testing, similar to the
+`TestProbe` used for asynchronous testing.
+
+
+### Spawning children
+
+With a name: 
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-child }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-child } 
+
+Anonymously:
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-anonymous-child }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-anonymous-child } 
+
+### Sending messages
+
+For testing sending a message a `TestInbox` is created that provides an `ActorRef` and methods to assert against the
+messages that have been sent to it.
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-message }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-message } 
+
+The `BehaviorTestkit` keeps track other effects you can verify, look at the sub-classes of `akka.typed.testkit.Effect`
+ 
+ * SpawnedAdapter
+ * Stopped
+ * Watched
+ * Unwatched
+ * Scheduled
+ 
+See the other public methods and API documentation on `BehaviourTestkit` for other typed of verification.
+   
+## Asynchronous testing
+
+Asynchronous testing uses a real `ActorSystem` that allows you to test your Actors in a realistic environment. 
+
+The minimal setup consists of the test procedure, which provides the desired stimuli, the actor under test, 
+and an actor receiving replies. Bigger systems replace the actor under test with a network of actors, apply stimuli 
+at varying injection points and arrange results to be sent from different emission points, but the basic principle stays 
+the same in that a single procedure drives the test.
+
+If you are familiar with testing untyped actors there are some differences:
+
+* No `ImplicitSender`, instead create `TestProbe[T]` that will be in the messages sent to your actor
```

Comment:
that's scala specific, but also there is no `testActor` and all assertions are on the probes instead